proposals/0002-gep.md

```diff
@@ -0,0 +1,216 @@
+---
+MEP: 0002
+Title: Unifed API
+Discussion: Link
+Implementation: Link
+---
+
+# Unified API
+
+## Abstract
+
+This GEP discusses the unified API that handles the interaction with LLM providers. This represents the feature set for MVP.
+
+## Motivation
+
+
+The Unified API abstracts much of the overhead of interacting with LLM providers.
+It stores all the provider specific logic in a single place and allowing developers to focus on the core logic. It also introduces durability to LLM apps. Additionally, unifying the API allows optimizations to be leveraged such as latency and cost improvements.
+
+
+### Requirements
+
+- R1: Handles all provider specific logic
+- R2: Easily maintained
+- R3: API schemas must unify common LLM request params (e.g. topP, N, etc)
+- R4: API routes must unify common LLM endpoints/API
+- R5: API schemas should allow to customize provider-specific params (e.g. a different messages for OpenAI/Anthropic)
+- R6: Focus on building out a single endpoint for /Chat. Most apps will be using this and then we can focus our efforts on making this part of our API super effective
+- R7: Disallow overrides in the MVP. All params are updated via config. This will change over time but I want users to tell which params they want to override (and how) instead of us making the assumption.
+
+
+## Design
+
+
+### Routes
+```yaml
+routes:
```

Comment:
@mkrueger12 when I look at these 5 routes it feels like we have a pattern there:

```
/v1/{domain/type}/{pool-id}/{everything else}/
```

Maybe then it's better to call new speech API in this fashion:

```
routes:
    chat: /v1/language/{pool-id}/chat/ 
    transcribers: /v1/speech/transcribers/{pool-id}/  
    speech-synthesizer:  /v1/speech/synthesizers/{pool-id}/ 
    multi-modal: /v1/multi/{pool-id}/multimodal/ 
    embedding: /v1/embeddings/{pool-id}/embed/ 
```